src/v2i-hub/ERVCloudForwardingPlugin/src/ERVCloudForwardingPlugin.cpp

```diff
@@ -35,43 +35,56 @@ namespace ERVCloudForwardingPlugin
     }
 
     void ERVCloudForwardingPlugin::RegisterRSULocation()
-    {        
-        try
+    {
+        uint32_t attempt = 0;
+        bool isRegistered = false;
+        while (attempt < _max_rsu_connection_attempt && !isRegistered)
         {
-            PLOG(logINFO) << "Create SNMP Client to connect to RSU. RSU IP:" << _rsuIp << ",\tRSU Port:" << _snmpPort << ",\tSecurity Name:" << _securityUser << ",\tAuthentication Passphrase: " << _authPassPhrase << endl;
-            auto snmpClient = std::make_shared<SNMPClient>(_rsuIp, _snmpPort, _securityUser, _authPassPhrase);
-            auto gps_sentense = snmpClient->SNMPGet(_GPSOID);
-            auto gps_map = ERVCloudForwardingWorker::ParseGPS(gps_sentense);
-            long latitude = 0;
-            long longitude = 0;
-            for (auto itr = gps_map.begin(); itr != gps_map.end(); itr++)
+            attempt++;
+            this_thread::sleep_for(chrono::seconds(1));
+            PLOG(logINFO) << "Attempting to register RSU " << attempt << " times." << endl;
+            try
             {
-                latitude = itr->first;
-                longitude = itr->second;
-            }
+                PLOG(logINFO) << "Create SNMP Client to connect to RSU. RSU IP:" << _rsuIp << ",\tRSU Port:" << _snmpPort << ",\tSecurity Name:" << _securityUser << ",\tAuthentication Passphrase: " << _authPassPhrase << endl;
+                auto snmpClient = std::make_shared<SNMPClient>(_rsuIp, _snmpPort, _securityUser, _authPassPhrase);
+                auto gps_sentense = snmpClient->SNMPGet(_GPSOID);
```

Comment:
Minor typo, should this be `gps_sentence`?

Reply:
https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/494/commits/b3688477f1099653f2f3cbdf19ee191834a8e458